src/main/java/com/squareup/kotlinpoet/ClassName.kt

```diff
@@ -109,7 +109,7 @@ class ClassName private constructor(
   }
 
   /** Returns the simple name of this class, like `"Entry"` for [Map.Entry].  */
-  fun simpleName() = names[names.size - 1]
+  fun simpleName() = names[names.size - 1].let { if (nullable) "$it?" else it }
```

Comment:
This isn't correct. We need this method to return the actual class name, nullable or not.